engine/planner_test.go

```diff
@@ -337,6 +314,59 @@ func TestPlan_ShutdownIdle(t *testing.T) {
 	}
 }
 
+// This test verifies that if the server capacity is
+// less than the pending count, including matrix builds,
+// and the server capacity is < the pool maximum,
+// additional servers are provisioned.
+//
+// However, this test should be called with the environment
+// variable "ENABLE_MATRIX_CALC" set to "true". As this is 
+// not easy to mock, it is currently disabled.
+// func TestPlan_MatrixMoreCapacity(t *testing.T) {
```

Comment:
Just set the env variable within this test case? Than you are able to test both scenarios.